### ISSUE TITLE
Log Weldr API fails

### ISSUE THREAD

**Issue by ondrejbudai** [osbuild/osbuild-composer#886]
Sometimes, things go south and we need to return 4xx or 5xx from the weldr API. I think it's a good idea to log these situations, so we can investigate them later. Sometimes, it isn't easy to get these logs from API consumers (see https://github.com/osbuild/cockpit-composer/issues/1061) so I think it would beneficial to log them also on the backend side.

**Comment by larskarlitski** [comment]
Not entirely sure about `4xx`, but `5xx` definitely. Would be cool to put together some guidelines about how and when we handle errors in general as part of this.

**Comment by bcl** [comment]
I agree with @larskarlitski  -- `4xx` should be things like user input errors, and they should all return something descriptive to the users so I'm not sure if it's very useful to log them.

**Comment by ondrejbudai** [comment]
I think I agree, we should log only `5xx`. I wonder if we can somehow differentiate the depsolving errors. Some are definitely bad requests (`4xx`, e.g. invalid combination of packages) but some are more on the internal server error side (`5xx`, e.g. composer cannot download metadata because of a broken mirror).

**Comment by bcl** [comment]
Is `5xx` appropriate for network errors, mirror failures, etc.? Those are going to happen quite a lot, and there is really nothing that can be done about them except to retry the request so those seems more like something that should be reported to the client and not logged.

I would expect a `5xx` to be more like a crash in the code, either a panic (indexing out of range, nil pointers, the like) or errors reading or writing to local disk (out of space, permissions, etc.) where a log entry can help debug the problem so it can be fixed for future releases.

**Comment by ondrejbudai** [comment]
It think 4xx shouldn't be used for network issues because these errors are "Client errors". I skimmed RFC7231 (HTTP 1.1) and it seems 500 is the most appropriate for this case:

> The 500 (Internal Server Error) status code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request.

I think faulty mirrors are an unexpected condition. I thought about 503 too but imho it's for situations when server is overloaded or a there's a scheduled maintenance
